azavea/echo-locator
# Add LogRocket tracking

## Overview

Adds the LogRocket library for usage and error tracking.

### Demo

Optional. Screenshots, `curl` examples, etc.
<img width="1440" alt="Screen Shot 2019-03-21 at 3 13 57 PM" src="https://user-images.githubusercontent.com/447977/54778806-00d78480-4bec-11e9-98c9-8cfd2adb88eb.png">

### Notes

- I ended up disabling network logging entirely. Some of the larger requests for data files were causing LogRocket to struggle, and the non-static requests all potentially need sanitization because they contain info that might be considered sensitive (voucher numbers, BHA user logins, and work / childcare addresses). Between these two factors, writing custom sanitization seemed like overkill and likely to remove most of the interesting requests the app was making, so I turned tracking off completely.
- It's not entirely possible to control LogRocket sessions from the frontend. So even if you call `identify()` with changed parameters, LogRocket often seems to continue the existing tracking session. It might be possible to work around this by clearing cookies or local storage, but it didn't seem worth it at this stage.
- We will need the client to create LogRocket accounts and provide us with the app IDs they want us to use. The lack of LogRocket app IDs shouldn't cause problems.
- I added code that I _think_ will insert the git commit into LogRocket so that we can track errors that occur on different versions of the app. I broke it into a separate commit so that it's easy to revert if it causes problems.

## Testing Instructions

 * First, test out the app immediately after checking out this branch to make sure that it functions properly without a defined `LOGROCKET_APP_ID`. You shouldn't see any console errors.
 * Pull in the changes in `env.yml.tmp` into your local `env.yml`, sign up for LogRocket, and create a LogRocket app. Put the LogRocket App ID into your `env.yml` in the appropriate spot. You should probably restart the dev server at this point to be safe.
 * Load the page and browse anonymously. Open your LogRocket dashboard and make sure you can see an Anonymous user session.
 * Load the page with a private browsing tab and log in. You should see a session with your username in LogRocket.


Resolves #57
